src/libtelio.udl

```diff
@@ -0,0 +1,572 @@
+    string fingerprint;
+    /// Heartbeat interval in seconds. Default value is 3600.
+    u32? heartbeat_interval;
+    /// Initial heartbeat interval in seconds. Default value is None.
+    u32? initial_heartbeat_interval;
+    /// QoS configuration for Nurse
+    FeatureQoS? qos;
+    /// Enable/disable collecting nat type
+    boolean? enable_nat_type_collection;
+};
+
+/// QoS configuration options
+dictionary FeatureQoS {
+    /// How often to collect rtt data in seconds. Default value is 300.
+    u32? rtt_interval;
+    /// Number of tries for each node. Default value is 3.
+    u32? rtt_tries;
+    /// Types of rtt analytics. Default is Ping.
+    sequence<string>? rtt_types;
+    /// Number of buckets used for rtt and throughput. Default value is 5.
+    u32? buckets;
+};
+
+/// Configurable features for Lana module
+dictionary FeatureLana {
+    /// Path of the file where events will be stored. If such file does not exist, it will be created, otherwise reused
+    string event_path;
+    /// Whether the events should be sent to produciton or not
+    boolean prod;
+};
+
+/// Enable wanted paths for telio
+dictionary FeaturePaths {
+    /// Enable paths in increasing priority: 0 is worse then 1 is worse then 2 ...
+    /// [PathType::Relay] always assumed as -1
+    sequence<PathType> priority;
+    /// Force only one specific path to be used.
+    PathType? force;
+};
+
+/// Configurable features for exit Dns
+dictionary FeatureExitDns {
+    /// Controls if it is allowed to reconfigure DNS peer when exit node is
+    /// (dis)connected.
+    boolean? auto_switch_dns_ips;
+};
+
+/// Enable meshent direct connection
+dictionary FeatureDirect {
+    /// Endpoint providers [default all]
+    EndpointProviders? providers;
+    /// Polling interval for endpoints [default 10s]
+    u64? endpoint_interval_secs;
+    /// Configuration options for skipping unresponsive peers
+    FeatureSkipUnresponsivePeers? skip_unresponsive_peers;
+};
+
+/// Avoid sending periodic messages to peers with no traffic reported by wireguard
+dictionary FeatureSkipUnresponsivePeers {
+    /// Time after which peers is considered unresponsive if it didn't receive any packets
+    u64 no_rx_threshold_secs;
+};
+
+/// Configure derp behaviour
+dictionary FeatureDerp {
+    /// Tcp keepalive set on derp server's side [default 15s]
+    u32? tcp_keepalive;
+    /// Derp will send empty messages after this many seconds of not sending/receiving any data [default 60s]
+    u32? derp_keepalive;
+    /// Enable polling of remote peer states to reduce derp traffic
+    boolean? enable_polling;
+    /// Use Mozilla's root certificates instead of OS ones [default false]
+    boolean use_built_in_root_certificates;
+};
+
+/// Whether to validate keys
+dictionary FeatureValidateKeys {
+    /// Defaults to true
+    boolean validate_keys;
+};
+
+/// Turns on connection resets upon VPN server change
+dictionary FeatureBoringtunResetConns {
```

Comment:
maybe this should just be a boolean?

Reply:
Yes, I updated these to not have to do custom conversion logic for UniFFI but of course it's better to keep the API simple than to avoid such conversions. Fixed